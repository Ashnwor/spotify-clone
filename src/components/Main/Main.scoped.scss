@import '../../styles/color-palette';

.main {
  position: relative;
  display: flex;
  overflow: auto;
  height: 0;
  flex: 1 1 auto;
  flex-direction: column;
  padding-top: 16px;
  padding-right: 32px;
  padding-left: 32px;
  background: $color-black-3;
  overflow-y: auto;
  row-gap: 32px;

  @media (max-width: 768px) {
    padding-right: 16px;
    padding-left: 16px;
  }
}

.loading-indicator {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -4px;
}
