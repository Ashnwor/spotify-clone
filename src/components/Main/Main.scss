.main {
  position: absolute;
  top: 60px;
  right: 0;
  bottom: 89px;
  left: 232px;
  background: #121212;
  overflow-y: scroll;
}

.main-padding {
  position: absolute;
  top: 16px;
  right: 32px;
  left: 32px;
  display: grid;
  height: 100%;
  grid-template-rows: repeat(10, 1fr);
  background: #121212;
  row-gap: 32px;
}

.loading-indicator {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -4px;
}
