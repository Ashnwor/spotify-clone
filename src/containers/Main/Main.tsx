import { useContext } from 'react';
import { ImpulseSpinner } from 'react-spinners-kit';
import styles from './Main.module.scss';
import Section from '../../components/Section/Section';
import Song from '../../types/Song';
import RecentlyPlayed from '../../context/RecentlyPlayed';

interface Props {
  songs: Song[];
}

const Main: React.FC<Props> = ({ songs }) => {
  const { recentlyPlayed } = useContext(RecentlyPlayed);

  return (
    <div className={styles.main}>
      {songs.length < 1 ? (
        <div className={styles['loading-indicator']}>
          <ImpulseSpinner frontColor="#1db954" />
        </div>
      ) : (
        <>
          <Section title="Songs" data={songs} />
          {recentlyPlayed && recentlyPlayed.length >= 1 && (
            <Section title="Recently Played" data={recentlyPlayed} noSeeAll />
          )}
        </>
      )}
    </div>
  );
};

export default Main;
