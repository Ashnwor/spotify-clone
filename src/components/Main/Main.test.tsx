import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

it('runs correctly', () => {
  const wrapper = shallow(<Main />);
  expect(wrapper).toMatchSnapshot();
});
