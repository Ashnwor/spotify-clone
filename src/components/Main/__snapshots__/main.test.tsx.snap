// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly 1`] = `
<div
  className="main"
>
  <div
    className="main-padding"
  >
    <Section
      data={Array []}
      title="Songs"
    />
  </div>
</div>
`;
